### PR TITLE
IGAPP-529: Send signal on app state change

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by SteffiStuffel — APPROVED** [review]
Tested on android emulator. Works fine. But i get some Error messages, not sure if they are cause by your changes.

Bildschirmfoto zu 2021-04-12 10-59-41

**Comment by klinzo** [comment]
> Tested on android emulator. Works fine. But i get some Error messages, not sure if they are cause by your changes.
> 
> Bildschirmfoto zu 2021-04-12 10-59-41

@SteffiStuffel thanks for reviewing. Never saw that errors before, did you maybe have some old dependencies and could retry after running `yarn install` please? Thanks. Will check for myself as well.

**Review by Taggotty — APPROVED** [review]
Not tested.

File: native/src/modules/app/components/AppStateListener.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  useEffect(() => {
+    AppState.addEventListener('change', handleAppStateChange)
+
+    return () => {
+      AppState.removeEventListener('change', handleAppStateChange)
+    }
+  }, [])
+
+  const handleAppStateChange = (nextAppState: string) => {
+    if (nextAppState === 'active') {
```
Comment (maxammann) [inline native/src/modules/app/components/AppStateListener.js:18]: there is also inactive on iOS: https://reactnative.dev/docs/appstate#app-states

I think this should also be tested on iOS and Android.

**Review by maxammann — APPROVED** [review]

**Comment by SteffiStuffel** [comment]
> > Tested on android emulator. Works fine. But i get some Error messages, not sure if they are cause by your changes.
> > Bildschirmfoto zu 2021-04-12 10-59-41
> 
> @SteffiStuffel thanks for reviewing. Never saw that errors before, did you maybe have some old dependencies and could retry after running `yarn install` please? Thanks. Will check for myself as well.

Can't reproduce it today, although I did not yarn install.

**Pull request merged**